### PR TITLE
Copy I18N assemblies when running without the linker

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/dotnet/aspnetcore/issues/20517

Also rename `MonoLinkerI18NAssemblies` -> `BlazorWebAssemblyI18NAssemblies`

### CODE REVIEW

**Comment by pranavkm** [comment]
I'm not sure what our tolerance is for taking build changes at this point. IMO, this should be a low risk change even for RC. It cleans up a property name (which is a bit optional) and copies additional assemblies to the output folder. I kept the implementation intentionally simple to avoid complicating our MSBuild targets.

**Review by SteveSandersonMS — APPROVED** [review]
The implementation looks great. I’d hope we can take this for ~~preview 5~~ RC. 

The tests look a bit quirky. If we’re copying all the i18n assemblies, not just the requested ones, could the tests assert we find more than just the one called ‘other’? Otherwise it looks like we do only copy the requested one. 

Also is it worth adding an assertion to one of the other linked-disabled tests to specify what is or is not copied in the default case when no option is specified?

File: src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets [hunk 1]
```diff
@@ -333,6 +333,14 @@
       <Output TaskParameter="Dependencies" ItemName="_BlazorResolvedAssemblyUnlinked" />
     </ResolveBlazorRuntimeDependencies>
 
+    <ItemGroup Condition="'$(BlazorWebAssembly18NAssemblies)' != 'none'">
+      <!--
+        Unless the user has asked for no-assemblies, copy all I18N assemblies to the build output.
+        We do not want to decipher the linker's format for passing in I18N options in our build targets
+      -->
+      <_BlazorResolvedAssemblyUnlinked Include="$(ComponentsWebAssemblyBaseClassLibraryPath)I18N*.dll" />
```
Comment (SteveSandersonMS) [inline src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets:341]: Is the following valid?
```suggestion
      <_BlazorResolvedAssemblyUnlinked Include="$(ComponentsWebAssemblyBaseClassLibraryPath)I18N.*.dll" />
```

Comment (pranavkm) [reply src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.targets:341]: There's a `I18N.dll` that is always required.

File: src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project>
 
   <PropertyGroup>
-    <MonoLinkerI18NAssemblies>none</MonoLinkerI18NAssemblies> <!-- See Mono linker docs - allows comma-separated values from: none,all,cjk,mideast,other,rare,west -->
+    <BlazorWebAssembly18NAssemblies>none</BlazorWebAssembly18NAssemblies> <!-- See Mono linker docs - allows comma-separated values from: none,all,cjk,mideast,other,rare,west -->
```
Comment (mkArtakMSFT) [inline src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.props:4]: This shouldn't be just BlazorWebAsembly**18N**Assemblies, make sure you add the `I` too:

Comment (mkArtakMSFT) [reply src/Components/WebAssembly/Build/src/targets/Blazor.MonoRuntime.props:4]: This is coming from `Internationalization` = `I` + 18 letters + `N` :)

**Review by mkArtakMSFT — CHANGES_REQUESTED** [review]
Let's get the name fixed before merging this.

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @pranavkm !

File: src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs [hunk 3]
```diff
@@ -276,6 +276,55 @@ public async Task Build_WithBlazorWebAssemblyEnableLinkingFalse_SatelliteAssembl
...
+            var buildOutputDirectory = project.BuildOutputDirectory;
+
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "I18N.dll");
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "I18N.Other.dll");
+            // When running without linker, we copy all I18N assemblies. Look for one additional
+            Assert.FileExists(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "I18N.West.dll");
+        }
+
+        [Fact]
+        public async Task Build_WithI8NOption_CopiesI8NAssembliesWithLinkerEnabled()
```
Comment (captainsafia) [inline src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs:305]: Nit: I would change this function name to reflect the fact that the behavior is a little different without the linker enabled (e.g. assemblies that are not set in `BlazorWebAssemblyI18NAssemblies` are not copied.

**Review by captainsafia — APPROVED** [review]
Left one small comment inline.

File: src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs (hunk 3 above)
Comment (pranavkm) [reply src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs:305]: We already verify that the test does not copy additional assemblies when you have the linker enabled. The fact that we include additional assemblies when it isn't is an implementation defect, I don't think we need to document it as part of the test name. We could very well improve it in the future if we get feedback.

**Pull request merged**